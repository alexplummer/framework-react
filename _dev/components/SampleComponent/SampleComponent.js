
// SampleComponent
// ============
// A basic sample component

// Imports
import React from 'react';

class SampleComponent extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="SampleComponent">
                <h4>Welcome to your new blank build!</h4>
                <p>Add in some Sass vars in setup.scss then create some components to get started.</p>
            </div>
        );
    }
}

// Exports
export default SampleComponent;
