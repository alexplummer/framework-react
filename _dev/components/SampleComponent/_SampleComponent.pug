
//- SampleComponent
//- ============
//- A basic sample component

//- Includes

mixin SampleComponent()
    
    .SampleComponent
