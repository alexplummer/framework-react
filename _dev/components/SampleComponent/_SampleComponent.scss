
// SampleComponent
// ============
// A basic sample component

.SampleComponent {
    @include shadow();
    text-align: center;
    background: #eee;
    border-bottom: 1px solid #ccc;
    border-radius: 10px;
    padding: $size-l;
    margin: $size-l;
}