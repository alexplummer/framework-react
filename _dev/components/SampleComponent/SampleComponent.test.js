
// SampleComponent
// ============
// A basic sample component

// Imports
import React from 'react';
import renderer from 'react-test-renderer';
import SampleComponent from './SampleComponent';

describe('SampleComponent renders correctly', () => {
    it('renders correctly', () => {
        const props = {};
        const rendered = renderer.create(
            <SampleComponent props={props} />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
