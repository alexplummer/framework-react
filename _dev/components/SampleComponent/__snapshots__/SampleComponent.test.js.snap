// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleComponent renders correctly renders correctly 1`] = `
<div
  className="SampleComponent"
>
  <h4>
    Welcome to your new blank build!
  </h4>
  <p>
    Add in some Sass vars in setup.scss then create some components to get started.
  </p>
</div>
`;
